tests/unit/experimental/pipeline/test_orchestrator.py

```diff
@@ -16,13 +16,76 @@
 from neo4j_genai.experimental.pipeline import Component
 from neo4j_genai.experimental.pipeline.pipeline import Orchestrator, Pipeline, RunStatus
 
+from tests.unit.experimental.pipeline.components import ComponentPassThrough
+
+
+def test_orchestrator_get_component_inputs_from_user_only() -> None:
+    pipe = Pipeline()
+    pipe.add_component(ComponentPassThrough(), "a")
+    pipe.add_component(ComponentPassThrough(), "b")
+    orchestrator = Orchestrator(pipe)
+    data = orchestrator.get_component_inputs("a", {}, {"a": {"value": "text"}})
+    assert data == {"value": "text"}
+    data = orchestrator.get_component_inputs("b", {}, {"a": {"value": "text"}})
+    assert data == {}
+
+
+def test_pipeline_get_component_inputs_from_parent_specific() -> None:
+    pipe = Pipeline()
+    pipe.add_component(ComponentPassThrough(), "a")
+    pipe.add_component(ComponentPassThrough(), "b")
+    pipe.connect("a", "b", input_config={"value": "a.result"})
+
+    # component "a" already run and results stored:
+    pipe._store.add("a", {"result": "a_result"})
+
+    orchestrator = Orchestrator(pipe)
+    data = orchestrator.get_component_inputs(
+        "b", {"value": "a.result"}, {"a": {"value": "text"}}
+    )
+    assert data == {"value": "a_result"}
```

Comment:
I think I'm confused now, is this not meant to return `"text"`?

Reply:
We're asking for the inputs for component "b", which must be the output of component "a".

Reply:
So does "text" not matter in this test?

Reply:
Indeed, it is not used. Would you prefer if it was deleted? I think it makes sense to test we're not messing up things when we have multiple components with same input names, but maybe there are other ways to test that?

Reply:
Indeed it's not used. I can remove it. I think it's good to test that we're not messing up with parameters when we have the same names in different locations, but I'll think about a way to test that explicitely.

Reply:
Added a new test [here](https://github.com/neo4j/neo4j-genai-python/pull/113/commits/6a026e1c802e7fe2177e00ba0eba4404d91ba37f#diff-a8ed923705bb0b915fbd58a296fc07729642c6d48b4d1e6157a6cccf02837705R103)